avocado/plugins/yaml_to_mux.py

```diff
@@ -417,6 +417,12 @@ def __iter__(self):
                    "variant": variant,
                    "mux_path": self.mux_path}
 
+    def __len__(self):
+        """
+        Report the number of variants
+        """
+        return len(self.variant_ids)
```

Comment:
OK, since `self.variant_ids` is set at `__init__` there should be no issues here.